File: rust-api-examples/examples/online_punctuation.rs
// Copyright (c) 2026  zengyw
//
// This file demonstrates how to use online punctuation with sherpa-onnx's Rust API.

use sherpa_onnx::{OnlinePunctuation, OnlinePunctuationConfig, OnlinePunctuationModelConfig};

fn main() {
    let config = OnlinePunctuationConfig {
        model: OnlinePunctuationModelConfig {
            cnn_bilstm: Some("./sherpa-onnx-online-punct-en-2024-08-06/model.onnx".to_string()),
            bpe_vocab: Some("./sherpa-onnx-online-punct-en-2024-08-06/bpe.vocab".to_string()),
            num_threads: 1,
            provider: Some("cpu".to_string()),
            ..Default::default()
        },
    };

    let punct = OnlinePunctuation::create(&config).expect("Failed to create OnlinePunctuation");

    let texts = [
        "how are you i am fine thank you",
        "The African blogosphere is rapidly expanding bringing more voices online in the form of commentaries opinions analyses rants and poetry",
    ];

    println!("----------");
    for text in texts {
        let out = punct
            .add_punctuation(text)
            .expect("Failed to add punctuation");

        println!("Input text: {text}");
        println!("Output text: {out}");
        println!("----------");
    }
}

File: dart-api-examples/add-punctuations/bin/online_punctuation.dart
// Copyright (c)  2026  Xiaomi Corporation
import 'dart:io';

import 'package:args/args.dart';
import 'package:sherpa_onnx/sherpa_onnx.dart' as sherpa_onnx;
import './init.dart';

void main(List<String> arguments) async {
  await initSherpaOnnx();

  final parser = ArgParser()
    ..addOption('model', help: 'Path to model.onnx')
    ..addOption('bpe-vocab', help: 'Path to bpe.vocab');

  final res = parser.parse(arguments);
  if (res['model'] == null || res['bpe-vocab'] == null) {
    print(parser.usage);
    exit(1);
  }

  final modelFile = res['model'] as String;
  final bpeVocab = res['bpe-vocab'] as String;
  final modelConfig = sherpa_onnx.OnlinePunctuationModelConfig(
    cnnBiLstm: modelFile,
    bpeVocab: bpeVocab,
    numThreads: 1,
    provider: 'cpu',
    debug: false,
  );

  final config = sherpa_onnx.OnlinePunctuationConfig(model: modelConfig);
  final punct = sherpa_onnx.OnlinePunctuation(config: config);

  final texts = [
    'how are you i am fine thank you',
    'The African blogosphere is rapidly expanding bringing more voices online in the form of commentaries opinions analyses rants and poetry',
  ];

  for (final t in texts) {
    final textWithPunct = punct.addPunct(t);
    print('----------');
    print('Before: $t');
    print('After: $textWithPunct');
  }
  print('----------');

  punct.free();
}
